settings.gradle.kts

```diff
@@ -13,44 +13,46 @@ plugins {
     id("org.gradle.toolchains.foojay-resolver-convention")
 }
 
-def gradleEnterpriseServer = "https://ge.opentelemetry.io"
-def isCI = System.getenv("CI") != null
-def geAccessKey = System.getenv("GRADLE_ENTERPRISE_ACCESS_KEY") ?: ""
+val gradleEnterpriseServer = "https://ge.opentelemetry.io"
+val isCI = System.getenv("CI") != null
+val geAccessKey = System.getenv("GRADLE_ENTERPRISE_ACCESS_KEY") ?: ""
 
-// if GE access key is not given and we are in CI, then we publish to scans.gradle.com
-def useScansGradleCom = isCI && geAccessKey.isEmpty()
+// if the GE access key is not given, and we are in CI, then we publish to scans.gradle.com
+val useScansGradleCom = isCI && geAccessKey.isEmpty()
 
 if (useScansGradleCom) {
     gradleEnterprise {
         buildScan {
             termsOfServiceUrl = "https://gradle.com/terms-of-service"
             termsOfServiceAgree = "yes"
-            uploadInBackground = !isCI
+            isUploadInBackground = !isCI
+
             publishAlways()
 
             capture {
-                taskInputFiles = true
+                isTaskInputFiles = true
             }
         }
     }
 } else {
     gradleEnterprise {
         server = gradleEnterpriseServer
+
         buildScan {
-            uploadInBackground = !isCI
+            isUploadInBackground = !isCI
 
-            publishIfAuthenticated()
-            publishAlways()
+            publishAlwaysIf(geAccessKey.isNotEmpty())
```

Comment:
I might be wrong, but I think there was a typo in the groovy version. If not, I can adjust.

Reply:
The two blocks definitely makes it harder to read. The differences between the two aren't huge, can they be combined?

Reply:
refactored into one block with checks inside